src/Server/BaseServer.ts

```diff
@@ -141,15 +142,66 @@ export class BaseServer {
    * Script object on the server (if it exists)
    */
   getScript(scriptName: string): Script | null {
+    const filename = sanitizeFilename(scriptName);
     for (let i = 0; i < this.scripts.length; i++) {
-      if (this.scripts[i].filename === scriptName) {
+      if (this.scripts[i].filename === filename) {
         return this.scripts[i];
       }
     }
 
     return null;
   }
 
+  /**
+   * @returns Array of all script filenames on server
+   */
+  getAllScriptFilenames(): string[] {
+    const filenames: string[] = [];
+    for (const file of this.scripts) {
```

Comment:
nit: could we just use a map?

Reply:
propbably, theres a lot of places in the codebase where a plain object or array is used when a Map would make a lot of sense

Reply:
yeah just figured might aswell fix the ones that your touching to keep your pr as small as possible its already huge 👍